src/merkle/tiered_smt/mod.rs

```diff
@@ -393,4 +448,22 @@ impl BottomLeaf {
         // TODO: hash in domain
         Rpo256::hash_elements(&elements)
     }
+
+    /// Returns contents of this leaf as a vector of (key, value) pairs.
+    ///
+    /// The keys are returned in their un-truncated form.
+    pub fn contents(&self, index: u64) -> Vec<(RpoDigest, Word)> {
+        self.values
+            .iter()
+            .map(|(key, val)| {
+                let key = RpoDigest::from([
+                    Felt::new(key[0]),
+                    Felt::new(key[1]),
+                    Felt::new(key[2]),
+                    Felt::new(index),
+                ]);
+                (key, *val)
+            })
+            .collect()
```

Comment:
I wouldn't `collect` here and leave to the user to build whatever type he wants (i.e. return `impl Iterator<Item = (RpoDigest, Word)>`)

That way, he can, for example, build a map, via:
`leaf.contents(15).collect::<BtreeMap<_, _>>()`

Reply:
This is primarily used inside another iterator, and it feels like returning an iterator within an iterator could be a bit complex.